Source/WebCore/css/parser/CSSParserImpl.cpp

```diff
@@ -1150,29 +1150,52 @@ void CSSParserImpl::consumeDeclarationListOrStyleBlockHelper(CSSParserTokenRange
         m_observerWrapper->skipCommentsBefore(range, true);
     }
 
-    auto consumeErrorUntilSemicolon = [&] () {
+    auto consumeUntilSemicolon = [&] {
         while (!range.atEnd() && range.peek().type() != SemicolonToken)
             range.consumeComponentValue();
     };
 
+    auto consumeNestedQualifiedRule = [&] {
+        auto rule = consumeQualifiedRule(range, AllowedRulesType::RegularRules);
+        if (!rule)
+            return false;
+        if (!rule->isStyleRule())
+            return false;
+        topContext().m_parsedRules.append(*rule);
+        return true;
+    };
+
     while (!range.atEnd()) {
         switch (range.peek().type()) {
         case WhitespaceToken:
         case SemicolonToken:
             range.consume();
             break;
         case IdentToken: {
-            const CSSParserToken* declarationStart = &range.peek();
+            const auto initialRange = range;
+            const auto declarationStart = &range.peek();
 
             if (useObserver)
                 m_observerWrapper->yieldCommentsBefore(range);
 
-            consumeErrorUntilSemicolon();
+            consumeUntilSemicolon();
 
-            consumeDeclaration(range.makeSubRange(declarationStart, &range.peek()), ruleType);
+            const auto declarationRange = range.makeSubRange(declarationStart, &range.peek());
+            auto valid = consumeDeclaration(declarationRange, ruleType);
```

Comment:
boolean variable are usually named with is/was/has or similar so this could be `isValidDeclaration`.